hpke/algs.go

```diff
@@ -44,7 +45,7 @@ func (k KEM) IsValid() bool {
 	}
 }
 
-func (k KEM) Scheme() kem.AuthScheme {
+func (k KEM) Scheme() kem.Scheme {
```

Comment:
What's the reason for this change?

Reply:
Since Auth* modes are optional, this allows to add, in the future, KEMs that do not support authentication.
Take a look in the updated version, now there are two functions `Scheme` and `AuthScheme`

Reply:
Hmm, this seems like it could bite us in the ass in the future. I would suggest we do one of two things. Either
1. We require that HPKE KEMs can be used securely for sender authentication (i.e., there is some sort of peer reviewed analysis of the KEMs security in this setting); or
2. We prevent the user from doing sender authentication unless the KEM is somehow marked as securely supporting this feature.

My preference would be (1.), since it (a) it's simpler and (b) it's likely to be case that any KEM used for encryption is also suitable for sender authentication.